### ISSUE TITLE
Update to consul.json does not trigger service restart

### ISSUE THREAD

**Issue by errygg** [johnbellone/consul-cookbook#381]
Had to update the node name to the hostname to support our deployment, but when the node name (i.e. node['consul']['config']['node_name']) attribute was overridden, it did not restart the consul service. Had to manually restart to capture the new node name.

Running on ubuntu jessie with chef client version 12.16.42.

**Comment by legal90** [comment]
Hi, @errygg 
This issue duplicates #307 and #341 

Service restart was disabled intentionally by https://github.com/johnbellone/consul-cookbook/commit/80f4a1ce4edb1638482e2391dabd1ba547ed7fad and https://github.com/johnbellone/consul-cookbook/pull/309
Please, refer to the discussion: https://github.com/johnbellone/consul-cookbook/issues/288 

TL;DR: Automatic restart of Consul servers could cause a cluster downtime when you run chef-client simultaneously on your nodes. Also, if Consul version has been changed (upgrade scenario), then Consul servers should be restarted manually, according to the [upgrade manual](https://www.consul.io/docs/upgrading.html).

**Comment by errygg** [comment]
Thanks @legal90! I guess I wasn't the first to see this. I'll go ahead and close the issue out. _Automate all the things (except consul restarts 😆 )_
